railties/lib/rails/app_loader.rb

```diff
@@ -10,13 +10,19 @@ module AppLoader # :nodoc:
     BUNDLER_WARNING = <<EOS
 Looks like your app's ./bin/rails is a stub that was generated by Bundler.
 
-In Rails #{Rails::VERSION::MAJOR}, your app's bin/ directory contains executables that are versioned
+Beginning In Rails 4, your app's bin/ directory contains executables that are versioned
 like any other source code, rather than stubs that are generated on demand.
 
-Here's how to upgrade:
+If your application was created in Rails 4+, you might be seeing this message
+if your bundle install command is using the --binstubs option and thereby overriding the
+standard Rails bin/rails binstub. You might need to stop using the --binstubs option,
+or at least ensure that it is using a path other than ./bin. You may need to coordinate
+this change with your environment's PATH.
```

Comment:
> you might need to stop using the --binstubs option, or at least ensure that it is using a path other than ./bin. You may need to coordinate this change with your environment's PATH.

I don't totally follow this.  The rest of the change is good. Can you tell me more about what exactly is going on with this though?